src/SFA.DAS.AssessorService.Web.Staff/Views/CertificateApprovals/_ApprovalsDetailsPartial.cshtml

```diff
@@ -1,131 +1,232 @@
+                                </p>
+                            </li>
+                            <li>
+                                <p>
+                                    <span class="bold">Apprenticeship start date: </span>
+                                    @{
+                                        var achievementStartDate = @results[i].LearningStartDate.HasValue ? @results[i].LearningStartDate.Value.ToString("dd/MM/yyyy") : "";
+                                    }
+                                    @achievementStartDate
+                                </p>
+                            </li>
+                            <li>
+                                <p>
+                                    <span class="bold">Achievement date: </span>
+                                    @{
+                                        var achievementDate = @results[i].AchievementDate.HasValue ? @results[i].AchievementDate.Value.ToString("dd/MM/yyyy") : "";
+                                    }
+                                    @achievementDate
+                                </p>
+                            </li>
+                            <li>
+                                <p>
+                                    <span class="bold">Certificate number:</span> @Html.DisplayFor(modelItem => @results[i].CertificateReference)
+                                </p>
+                            </li>
+                        </ul>
+
+                        <h2 class="heading-medium">Postage details: </h2>
+                        <p>
+                            The certificate was posted to this address.
+                        </p>
+                        <ul>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactName)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactOrganisation)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactAddLine1)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactAddLine2)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactAddLine3)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactAddLine4)
+                            </li>
+                            <li>
+                                @Html.DisplayFor(modelItem => @results[i].ContactPostCode)
+                            </li>
+                        </ul>
+                    </div>
+                </td>
+            </tr>
+        }
     </tbody>
 </table>
```

Comment:
We need to discuss how to make this better.  7 Params per certificate is only going to allow 7(ish) certificates to be posted and that's with the param limit upped to 50.  Pagination is also not the answer.

My first idea is to not just rely on a form POST for this page, but to use JS to build up some json to POST to the server.

OR each of those approvalresults[n] params combined into one like "00010844|blah|blah"

At the very least, need to review whether each of these params is absolutely necessary. For example .IsApproved is duplicated in each element of the array and actionHint and pageIndex.

`approvalresults[0].IsApproved: ToBeApproved
approvalresults[0].CertificateReference: 00010844
approvalresults[0].IsApproved: ToBeApproved
approvalresults[0].PrivatelyFundedStatus: 
actionHint: SentForApproval
pageIndex: 1
approvalresults[0].ReasonForChange: 
approvalresults[1].IsApproved: ToBeApproved
approvalresults[1].CertificateReference: 00010843
approvalresults[1].IsApproved: ToBeApproved
approvalresults[1].PrivatelyFundedStatus: 
actionHint: SentForApproval
pageIndex: 1
approvalresults[1].ReasonForChange: `

Reply:
We will release the code "as-is" and then handle the WAF restictions by re-working the Staff UI screen to make use of Javascript and application/json POST for the data elements that are to be changed.